driver/src/contracts/stablex_auction_element.rs

```diff
@@ -0,0 +1,299 @@
+use std::collections::HashMap;
+
+use byteorder::{BigEndian, ByteOrder};
+use web3::types::{H160, U256};
+
+use dfusion_core::models::{BatchInformation, Order};
+
+use crate::util::u128_to_u256;
+
+#[derive(Debug, Default, PartialEq)]
+pub struct StableXAuctionElement {
+    valid_from: U256,
+    valid_until: U256,
+    pub sell_token_balance: u128,
+    pub order: Order,
+}
+
+impl StableXAuctionElement {
+    pub fn in_auction(&self, index: U256) -> bool {
+        self.valid_from <= index && index <= self.valid_until
+    }
+
+    pub fn from_bytes(order_count: &mut HashMap<H160, u16>, bytes: &[u8; 113]) -> Self {
+        let account_id = H160::from(&bytes[0..20]);
+
+        // these go together (since sell_token_balance is emitted as u256 and treated as u128
+        let sell_token_balance = BigEndian::read_u128(&bytes[36..52]);
+        let sell_token_balance_padding = BigEndian::read_u128(&bytes[20..36]);
+        assert_eq!(
+            sell_token_balance_padding, 0,
+            "User has too large balance to handle."
+        );
+
+        let buy_token = u16::from_le_bytes([bytes[53], bytes[52]]);
+        let sell_token = u16::from_le_bytes([bytes[55], bytes[54]]);
+        let valid_from = U256::from(u32::from_le_bytes([
+            bytes[59], bytes[58], bytes[57], bytes[56],
+        ]));
+        let valid_until = U256::from(u32::from_le_bytes([
+            bytes[63], bytes[62], bytes[61], bytes[60],
+        ]));
+        let is_sell_order = bytes[64] > 0;
+        let numerator = BigEndian::read_u128(&bytes[65..81]);
+        let denominator = BigEndian::read_u128(&bytes[81..97]);
+        let amount = BigEndian::read_u128(&bytes[97..113]);
+        let (buy_amount, sell_amount) =
+            compute_buy_sell_amounts(numerator, denominator, amount, is_sell_order);
+        let order_counter = order_count.entry(account_id).or_insert(0);
+        *order_counter += 1;
+        StableXAuctionElement {
+            valid_from,
+            valid_until,
+            sell_token_balance,
+            order: Order {
+                batch_information: Some(BatchInformation {
+                    slot_index: *order_counter - 1,
+                    slot: U256::from(0),
+                }),
+                account_id,
+                buy_token,
+                sell_token,
+                buy_amount,
+                sell_amount,
+            },
+        }
+    }
+}
+
+fn compute_buy_sell_amounts(
+    numerator: u128,
+    denominator: u128,
+    remaining: u128,
+    is_sell_order: bool,
+) -> (u128, u128) {
+    // TODO - use u128_to_u256 for overflow handling (this is a conversion used in naive_solver)
+    assert!(
+        remaining <= denominator,
+        "Smart contract should never allow this inequality"
+    );
+    let mut other = 0 as u128;
```

Comment:
Why is other defined outside of the if statement? Also isn't it the buy amount? Can we name it more appropriately.

Reply:
It was put outside so that the else branch of the if statement could also treat it as the `sell_amount`. Of course, since this remains unimplemented, it does not need to be here at the moment.